### PR TITLE
Closes #1882 adds a toggle to display past year or full data

### LINKED ISSUE #1882: Feature: Prevent horizontal scrollbar in Flags Over Time table.
## Feature Details
Display full width of table so user doens't have to scroll to the bottom to scroll to the right.

## Possible Implementation
Or freeze the first three columns and add a scroll bar at the top, but that seems difficult.

## Additional Comments
image

### PR DESCRIPTION
**Pull request by zdz2101**
## Overview
Create multiple table/gt objects in `Widget_FlagOverTime()` to then feed in to the JS script
Modify JS script toggle between the two objects 

Reprex:
```
devtools::install_github("Gilead-Biostats/gsm", ref = "fix-1882")
library(dplyr)
library(gsm)

reportingResultsSubset <- dplyr::filter(
  reportingResults,
  GroupID %in% head(unique(reportingResults$GroupID))
)
reportingResultsSubset2 <- reportingResultsSubset %>%
  mutate(SnapshotDate = SnapshotDate + 365)

reportingResultsSubset3 <- bind_rows(
  reportingResultsSubset2,
  reportingResultsSubset
)

Widget_FlagOverTime(
  dfResults = reportingResultsSubset3,
  dfMetrics = reportingMetrics
)
```

image
image

### CODE REVIEW

File: R/Widget_FlagOverTime.R
Comment (zdz2101) [inline R/Widget_FlagOverTime.R:41]: Curious what if snapshots are collected biweekly, quarterly, different frequencies, is there a more "flexible" way to be robust for the user?

Comment (lauramaxwell) [reply R/Widget_FlagOverTime.R]: in general, we collect monthly snapshots, but even if it were collected weekly, this would provide roughly a quarter-long view, so i think its a reasonable duration to show as the default

File: inst/htmlwidgets/Widget_FlagOverTime.js
Comment (lauramaxwell) [inline inst/htmlwidgets/Widget_FlagOverTime.js:18]: maybe try to implement the toggle switch, like we use in Report_StudyInfo.R (lines 37-44)

**Comment by lauramaxwell** [comment]
Not sure exactly what the issue is, but the dropdown filter menu is no longer filtering the table properly

**Comment by lauramaxwell** [comment]
Was still seeing an issue with the subset dropdown. part of it had to do with hex code changes in rbm-viz, but haven't totally fixed it. see most recent commit to see what I mean. I'm also ok with pushing this to another PR if it's out of scope here.

**Comment by lauramaxwell** [comment]
Figured out the last piece to be able to close #1935 via js. I think everything is looking good now, but @zdz2101 take a look!

File: inst/htmlwidgets/Widget_FlagOverTime.js
Comment (zdz2101) [reply inst/htmlwidgets/Widget_FlagOverTime.js:18]: I think I tried to go somewhere along those lines of just doing it solely in R but I couldn't get past the fact the js widget at its current state only takes 1 table to display, not sure how it would work otherwise @samussiah any thoughts?

Comment (lauramaxwell) [reply inst/htmlwidgets/Widget_FlagOverTime.js:18]: yeah, at the moment its just a default styled button, but it works as expected. elsewhere this toggle button is inserted via html, so i'm not clever enough to quickly figure out how to do that here in js. we can add another issue if folks are annoyed by how it looks.

Comment (lauramaxwell) [reply inst/htmlwidgets/Widget_FlagOverTime.js]: i actually was able to get like 80% of the way there- got the switch to appear with the label next to it, but it doesn't slide when clicked. i think it's probably worth figuring this out tomorrow before merging this in. @jonthegeek or @samussiah - any tips? 
image

Comment (lauramaxwell) [reply inst/htmlwidgets/Widget_FlagOverTime.js]: got it to slide when clicked!! i'll let someone else take a look before i approve though

File: inst/htmlwidgets/Widget_FlagOverTime.js
Comment (samussiah) [reply inst/htmlwidgets/Widget_FlagOverTime.js:75]: This should do the same thing.  Love the toggle!

**Review by lauramaxwell — APPROVED** [review]
looking good! since i made edits, i won't merge, but i'm done fiddling with the toggle and i think this is ready to go!

**Comment by zdz2101** [comment]
> looking good! since i made edits, i won't merge, but i'm done fiddling with the toggle and i think this is ready to go!

Certainly improved from before, works for me

**Review by samussiah — APPROVED** [review]
Nice work!

**Pull request merged**
